File: removing_stars_from_a_string.java
import java.util.Stack;

public class removing_stars_from_a_string {

    public static void main(String[] args) {
        Solution s = new Solution();
        s.removeStars("leet**cod*e");
    }
}

class Solution {
    public String removeStars(String s) {
        Stack<Character> st = new Stack<>();
        int n = s.length();
        for (int i = 0; i < n; i++) {

            if (Character.isAlphabetic(s.charAt(i))) {
                st.push(s.charAt(i));
            } else {
                if (!st.isEmpty())
                    st.pop();
            }
        }
        StringBuilder sb = new StringBuilder();
        while (!st.isEmpty()) {
            sb.append(st.peek());
            st.pop();
        }

        System.out.println(sb.reverse().toString());

        return "";
    }
}

File: removing_stars_from_a_string.cpp
#include <bits/stdc++.h>

using namespace std;
class Solution
{
public:
    string removeStars(string s)
    {
        int star = 0;
        int nonStar = -1;
        int n = s.size();
        while (star < n)
        {

            if (isalpha(s.at(star)))
            {
                star++;
            }
            else if (!isalpha(s.at(star)))
            {
                s[star] = '2';
                int i = star;
                while (i >= 0 and !isalpha(s[i]))
                {
                    i--;
                }
                s[i] = '2';
                star++;
            }
        }
        string ans = "";
        for (int i = 0; i < n; i++)
        {
            if (isalpha(s.at(i)))
                ans += s.at(i);
        }
        return ans;
    }
};
int main()
{

    string str = "leet**cod*e";

    Solution s = Solution();
    cout << s.removeStars(str) << endl;

    return 0;
}